lib/index.d.ts

```diff
@@ -0,0 +1,43 @@
+import {Request} from 'express';
+import * as passport from 'passport';
+
+export interface StrategyOptions {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  passReqToCallback?: true;
+  scope?: string[];
+}
+
+export interface StrategyOptionsWithRequest {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  passReqToCallback: true;
+  scope?: string[];
+}
+
+export interface VerifyOptions {
+  message: string;
+}
+
+export interface VerifyCallback {
+  (error: any, user?: any, options?: VerifyOptions): void;
```

Comment:
This defines a function interface, it could have been done as an old school `export type` but for whatever reason I decided an interface would be more clear when writing these. The VerifyCallback passes an `error` which can be of any type (meaning either a JS object with any form, or any primative such as a number, string, boolean, etc). It's often best not to assume with error's because different modules do different things. In this case it is likely to be a standard Error object but it's not in the docs so I left it as `any` for now.

Generally typing something as `any` isn't something you want to stay with unless it's typing someone elses module because it essentially lets anything through that type which as you may have guessed is sort of like saying "I don't really care" to the type system.

The user parameter is optional.

The options value is an object which has a "message" parameter of type string as you can see in the above interface.